syntax_test_fortran.F90

```diff
@@ -118,13 +118,20 @@
 !
    'This is a simple string'
 !  ^^^^^^^^^^^^^^^^^^^^^^^^^ string.quoted.single.fortran
+!  ^ punctuation.definition.string.begin.fortran
+!                          ^ punctuation.definition.string.end.fortran
    &
 !  ^ punctuation.separator.continuation.fortran
 !
    'This is a simple string,  &
    & but "with" continuation, &
    & and more continuation!'
 !  ^ punctuation.separator.continuation.fortran
+!
+   'Line continuation character missing on the next line, & ! comment
+      but highlighting of the rest of the file is still not messed up.'
```

Comment:
Perhaps we should not color-code comments in this case.

The string
```fortran
      teststring = 'gives warning on compile & 
                         but is printed correctly'
```
compiles with a warning using gfortran 9.3.0, but is printed correctly. With a side-comment, I get an error:
```fortran
      teststring = 'does not & ! comment 
                         compile'
```
 gives
```shell
  312 |       teststring = 'does not & ! comment
      |                    1
Error: Unterminated character constant beginning at (1)
```

Reply:
You're right, comments after line continuation seem to be possible after other expressions, but I didn't test it for strings separately, where they indeed result in a compiler error. I thought I needed to consume the rest of the line in the regex match to avoid to pop the anonymous context in `string-continuation` if the `&` is not the last character in the line. But the cause of the error seems to be a missing quote at the end of the string, according to the error message. In fact, the `&` in the first line of the string seems to only be treated as a line continuation character if it's not followed by any non-whitespace character (but comments are allowed as separate lines):

```fortran
program test
    character(len=100) :: teststring1
    character(len=100) :: teststring2

    teststring1 = 'text & ! no comment'

    teststring2 = 'text &
        ! comment
        &and more text'

    print*, teststring1
    print*, teststring2

end program
```

gives me:
```
 text & ! no comment
 text and more text
```

I will try to fix it.

***

Btw, I noticed some other small issues, for example variable names like `exitflag` partly highlighted as keyword due to a missing word boundary in the corresponding regex. And some of the regexes could be improved in general, to not scope whitespace characters and remove some lookbehinds. If I open more PRs, should I still target the development branch or the main branch of this repo?

Reply:
The correct detection of line continuation in strings should be fixed now with 2a23b47.

Reply:
> If I open more PRs, should I still target the development branch or the main branch of this repo?

We wish to try to follow gitflow. You can target main for PRs with bugfixes to existing functionality. For new added features, the PR can target development. I see the comment in the README is not very helpful here.

Thanks for all these contributions! Great work - I will review them as fast as I can.